app/controllers/users_controller.rb

```diff
@@ -115,6 +115,15 @@ def profile
     @notes = DrupalNode.page(params[:page])
                        .order("nid DESC")
                        .where(type: 'note', status: 1, uid: @user.uid)
+                       .joins(:drupal_tag)
+                       .where('term_data.name NOT LIKE ?', 'question:%')
+                       .group('node.nid')
```

Comment:
Ah, i see. I believe grouping by node.nid ought to then show only one record per node, and so this looks OK in that respect. But are you joining the right table?  It also seems right; see: https://github.com/publiclab/plots2/blob/master/app/controllers/tag_controller.rb#L9

Let me think. If it's inconsistent, perhaps it works when the node has only a single tag and it's `question:%` but if it has other tags, then the  `question:%` is excluded but they get included anyways due to the other tags. I've seen this before.

So you need two queries, I believe. One which collects the `nids` to exclude, and a second which queries with those `nids` excluded. A similar question came up in this discussion: https://groups.google.com/forum/#!searchin/plots-dev/exclude/plots-dev/HUfgxx5kwPA/Z3pQhR1fDwAJ

The "EXCEPT" query should've worked, or we could have looked further into using a subquery, but for expediency, I just wrote simpler code that was not quite as efficient: https://github.com/publiclab/plots2/issues/375

Consider that in my `NOT IN` code, you are holding the total # of nids that you'd like to exclude in memory, so try to narrow your first query well so as not to collect ALL nids every time.